internal/environment/environment.go

```diff
@@ -0,0 +1,161 @@
+// The environment package contains probing functionality to
+// detect information on the Kubernetes cluster environment.
+package environment
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/discovery"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	manifestsv1alpha1 "package-operator.run/apis/manifests/v1alpha1"
+)
+
+var _ manager.Runnable = (*Manager)(nil)
+
+const (
+	environmentProbeInterval    = 1 * time.Minute
+	openShiftClusterVersionName = "version"
+	openShiftProxyName          = "cluster"
+)
+
+type Sink interface {
+	InjectEnvironment(env manifestsv1alpha1.PackageEnvironment)
+}
+
```

Comment:
```suggestion
type Sinker interface {
	SetEnvironment(env manifestsv1alpha1.PackageEnvironment)
}

type Sink struct {
    env  manifestsv1alpha1.PackageEnvironment
    lock sync.RWMutex
}

func (s *Sink) SetEnvironment(env manifestsv1alpha1.PackageEnvironment) {
	s.lock.Lock()
	defer s.lock.Unlock()
	s.env = *env.DeepCopy()
}

func (s *Sink) GetEnvironment() manifestsv1alpha1.PackageEnvironment {
    s.lock.RLock()
	defer s.lock.RUnlock()
	return *s.env.DeepCopy()
}
```

Very optional, but embedding an implementation of this interface into reconcilers probably makes sense as the behavior would just be duplicated otherwise.